### ISSUE TITLE
[FR]Do you plan to add DOH server as upstream for unbound？

### ISSUE THREAD

**Issue by LittleGrocery** [NLnetLabs/unbound#525]
**Current behavior**
Is there a current behavior that the feature relates to?
If yes, would you wish the current behavior to change?

Now unbound can only use Dot（@853） ， dns（@53）as upstream server
 

**Describe the desired feature**
A clear and concise description of what the feature should be.

Want to use Doh（@443）as upstream server

**Potential use-case**
Describe how you see this feature being useful to other Unbound users.
I know in many cases Dot performance are better than Doh.  However, ， in some network environment（like me），Doh is fast and better.

**Comment by yan12125** [comment]
Seems the same feature request as https://github.com/NLnetLabs/unbound/issues/308?

**Comment by LittleGrocery** [comment]
> Seems the same feature request as #308?

Maybe one year passed. The developer still doesn't want to implement this feature.

**Comment by karson33** [comment]
I guess one reason why DOH server as upstream is not deem urgent is because most, if not all, root/TLD servers themselves are not DOH compatible. Even Unbound can support DOH upstream, there is no root/TLD server DOH to work with. 
If you are using DOH like Google and Cloudflare etc as upstream, then the use case of Unbound is less significant (instead of unbound, you may consider to set up a DOH "forwarder" using one of the below methods:
https://github.com/kpadron/doh-forwarder/
https://www.sindastra.de/p/1332/how-to-set-up-doh-dns-over-https-forwarder
https://www.nginx.com/blog/using-nginx-as-dot-doh-gateway/

**Comment by agneevX** [comment]
> I guess one reason why DOH server as upstream is not deem urgent is because most, if not all, root/TLD servers themselves are not DOH compatible

Not really. It's not like nameservers support DNS-over-TLS, but that has already been implemented.

**Comment by PeterDaveHello** [comment]
> Seems the same feature request as #308?

Looks so!

> Maybe one year passed. The developer still doesn't want to implement this feature.

In the case, opening another issue might not help, maybe just use the original issue to discuss with the maintainers? Maybe they just don't have enough bandwidth to implement it, not like they don't want to, at least, in #308, they said:

> If you wanted DOH towards upstream servers, then this is a new feature. And it should wait for the already referenced DoH implementation, and also for socket reuse changes that are upcoming. It would be prudent to then first resolve bugs in that code. And then start thinking about this feature.

**Comment by p1r473** [comment]
I have it successfully using DOT (DNS over TLS) which is great
```#@853 at the end of the primary and secondary server IPs tells unbound to connect to Cloudflare using port 853, which is the secured end point for the service.
#forward-ssl-upstream: yes - is the instruction to use DNS over TLS, in this case for all queries (name: “.”)
server:
    tls-cert-bundle: /etc/ssl/certs/ca-certificates.crt
forward-zone:
    name: "."
    forward-tls-upstream: yes

    # Cloudflare DNS
    forward-addr: 1.1.1.1@853#cloudflare-dns.com
    forward-addr: 1.0.0.1@853#cloudflare-dns.com
    #forward-addr: 2606:4700:4700::1111@853#cloudflare-dns.com
    #forward-addr: 2606:4700:4700::1001@853#cloudflare-dns.com

    # NordVPN
    #forward-addr: 103.86.96.100@853#dns1.nordvpn.com
    #forward-addr: 103.86.99.100@853#dns2.nordvpn.com

    # Quad9
    #forward-addr: 2620:fe::fe@853#dns.quad9.net
    #forward-addr: 9.9.9.9@853#dns.quad9.net
    #forward-addr: 2620:fe::9@853#dns.quad9.net
    #forward-addr: 149.112.112.112@853#dns.quad9.net
```

**Comment by stm32repo** [comment]
Strange as it may seem, but the speed of DoH is much higher than DoT. I am currently using dnscrypte-proxy.

**Comment by ninetailedtori** [comment]
The other issue with DoT upstream is it's often port-blocked by many servers, due to many general-use-case security models. To be able to forward to upstream DoH would be more useful until TLS/QUIC is adopted more fully.

**Comment by baudneo** [comment]
Would just like to throw my vote in to use upstream DoH servers.

**Comment by gthess** [comment]
Closing this as duplicate of #308.
